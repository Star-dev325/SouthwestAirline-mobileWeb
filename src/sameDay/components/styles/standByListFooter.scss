.stand-by-list-footer {
  &--sub-note {
    @include bgpblue;
    @include fullwidth;
    @include white;
  }

  &--refund-taxes-note {
    @include bggreen;
    @include fullwidth;
    @include white;
    white-space: break-spaces;

    &_right {
      @include align-right;
      @include bggreen;
      @include fullwidth;
      @include white;
      white-space: break-spaces;
    }
  }

  &--nav {
    @include border-none;
    @include p3;
  }

  &--nav_webview {
    padding-bottom: 5rem;
  }
}

.stand-by-list-footer--price-total {
  @include bgpblue;
  @include p5;
  @include white;

  .price-line--title {
    @include align-middle;
    @include lineheight14;
    width: 45%;
  }

  .price-amount {
    @include align-bottom;
    @include larger;
  }

  .price-credit-message {
    @include align-right;
  }

  .price-line--currency_total-per-section {
    @include white;
    @include xxxlarge;
  }

  .stand-by-list-footer--price-refund-message {
    @include align-right;

    &_top-margin {
      @include mt3;
    }
  }
}

.stand-by-list-footer--btn {
  @include bgpblue;
  @include pb5;
  @include px5;
  @include white;

  &_right {
    @include p5;
  }
}

.stand-by-list-footer--taxes-and-fees {
  @include bgpblue;
  @include flex;
  @include px5;
  @include white;
}

.stand-by-list-footer--taxes-and-fees_top {
  @include flex;
  @include flex-column;
}

.stand-by-list-footer--taxes-and-fees_reverse {
  @include flex;
  @include flex-column-reverse;
}

.stand-by-list-footer--divider {
  @include bgpblue;
  @include py3;

  &-rule {
    opacity: 0.1;
  }
}
