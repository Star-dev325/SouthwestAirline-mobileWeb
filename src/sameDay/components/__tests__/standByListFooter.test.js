import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { StandByListFooter } from 'src/sameDay/components/standByListFooter.jsx';
import sameDayPurchaseConfirmationBuilder from 'test/builders/apiResponse/sameDayPurchaseConfirmationPageBuilder';
import configureMockStore from 'test/unit/helpers/configureMockStore';

describe('StandByListFooter', () => {
  it('should  render standBy list footer component with all information', () => {
    const {
      sameDayConfirmationPage: {
        fareSummary: {
          total,
          total: { item: title },
          taxesAndFeesWithLinks
        },
        _links: {
          enhancedStandbyList: { labelText: standbyLabel }
        }
      }
    } = new sameDayPurchaseConfirmationBuilder().build();

    const { container } = createComponent(
      { isShowPoints: false, standbyLabel, taxesAndFeesWithLinks, title, total },
      {}
    );

    expect(container).toMatchSnapshot();
  });

  it('should render total with points and tax in the standBy list footer component', () => {
    const {
      sameDayConfirmationPage: {
        fareSummary: {
          total: pointsTotal,
          total: { item: title },
          taxesAndFeesWithLinks,
          totalPointsTax: total
        },
        _links: {
          enhancedStandbyList: { labelText: standbyLabel }
        }
      }
    } = new sameDayPurchaseConfirmationBuilder().withPointsAmountDue().build();

    const { container } = createComponent(
      { isShowPoints: true, pointsTotal, standbyLabel, taxesAndFeesWithLinks, title, total },
      {}
    );

    expect(container).toMatchSnapshot();
  });

  it('should render taxCreditRefund with points credit and tax credit in the standBy list footer component', () => {
    const {
      sameDayConfirmationPage: {
        fareSummary: {
          refundMessage,
          total: pointsTotal,
          totalCreditPointsTax: taxCreditRefund,
          taxesAndFeesWithLinks,
          totalCreditPointsTax: { item: taxTitle },
          totalPointsTax: total
        },
        _links: {
          viewBoardingPositions: { labelText: standbyLabel }
        }
      }
    } = new sameDayPurchaseConfirmationBuilder().withPointsAndTaxCredit().build();

    const { container } = createComponent(
      {
        isShowPoints: true,
        pointsTotal,
        refundMessage,
        taxCreditRefund,
        standbyLabel,
        taxesAndFeesWithLinks,
        taxTitle,
        total
      },
      {}
    );

    expect(container).toMatchSnapshot();
  });

  it('should render total with points and no tax in the standBy list footer component', () => {
    const {
      sameDayConfirmationPage: {
        fareSummary: {
          total,
          total: { item: title },
          taxesAndFeesWithLinks
        },
        _links: {
          enhancedStandbyList: { labelText: standbyLabel }
        }
      }
    } = new sameDayPurchaseConfirmationBuilder().withPointsNoTaxAmountDue().build();

    const { container } = createComponent(
      { isShowPoints: true, standbyLabel, taxesAndFeesWithLinks, title, total },
      {}
    );

    expect(container).toMatchSnapshot();
  });

  it('should render Boarding details label in the standBy list footer component', () => {
    const {
      sameDayConfirmationPage: {
        fareSummary: {
          refundMessage,
          totalCredit,
          totalCredit: { item: title },
          taxesAndFeesWithLinks
        },
        _links: {
          viewBoardingPositions: { labelText: viewBoardingPositionsLabel }
        }
      }
    } = new sameDayPurchaseConfirmationBuilder().withTotalCreditDue().build();

    const { container } = createComponent(
      { refundMessage, taxesAndFeesWithLinks, title, total: totalCredit, totalCredit, viewBoardingPositionsLabel },
      {}
    );

    expect(container).toMatchSnapshot();
  });

  it('should not render standby list button in the standBy list footer component', () => {
    const {
      sameDayConfirmationPage: {
        fareSummary: {
          total,
          total: { item: title },
          taxesAndFeesWithLinks
        }
      }
    } = new sameDayPurchaseConfirmationBuilder().build();

    const { container } = createComponent({ taxesAndFeesWithLinks, total, title }, {});

    expect(container).toMatchSnapshot();
  });

  it('should not render total price in the standBy list footer component', () => {
    const {
      sameDayConfirmationPage: {
        fareSummary: {
          total: { item: title },
          taxesAndFeesWithLinks
        },
        _links: {
          enhancedStandbyList: { labelText: standbyLabel }
        }
      }
    } = new sameDayPurchaseConfirmationBuilder().build();

    const { container } = createComponent({ standbyLabel, taxesAndFeesWithLinks, title }, {});

    expect(container).toMatchSnapshot();
  });

  it('should render total credit and standby list button in the standBy list footer component ', () => {
    const { sameDayConfirmationPage } = new sameDayPurchaseConfirmationBuilder().withTotalCreditDueStandBy().build();
    const refundMessage = sameDayConfirmationPage.fareSummary.refundMessage;
    const standbyLabel = sameDayConfirmationPage._links.enhancedStandbyList.labelText;
    const taxesAndFeesWithLinks = sameDayConfirmationPage.fareSummary.taxesAndFeesWithLinks;
    const title = sameDayConfirmationPage.fareSummary.totalCredit.item;
    const totalCredit = sameDayConfirmationPage.fareSummary.totalCredit;

    const { container } = createComponent(
      { refundMessage, standbyLabel, taxesAndFeesWithLinks, title, totalCredit },
      {}
    );

    expect(container).toMatchSnapshot();
  });

  it('should render standby list button with extra bottom padding in the standBy list footer component when isWebView', () => {
    const { sameDayConfirmationPage } = new sameDayPurchaseConfirmationBuilder().withTotalCreditDueStandBy().build();
    const refundMessage = sameDayConfirmationPage.fareSummary.refundMessage;
    const standbyLabel = sameDayConfirmationPage._links.enhancedStandbyList.labelText;
    const taxesAndFeesWithLinks = sameDayConfirmationPage.fareSummary.taxesAndFeesWithLinks;
    const title = sameDayConfirmationPage.fareSummary.totalCredit.item;
    const totalCredit = sameDayConfirmationPage.fareSummary.totalCredit;

    const { container } = createComponent(
      { refundMessage, standbyLabel, taxesAndFeesWithLinks, title, totalCredit, isWebView: true },
      {}
    );

    expect(container).toMatchSnapshot();
  });

  it('should not render explanatory note(taxesAndFeesWithLinks) in the standBy list footer component', () => {
    const { sameDayConfirmationPage } = new sameDayPurchaseConfirmationBuilder().build();
    const standbyLabel = sameDayConfirmationPage._links.enhancedStandbyList.labelText;
    const title = sameDayConfirmationPage.fareSummary.total.item;
    const total = sameDayConfirmationPage.fareSummary.total;

    const { container } = createComponent({ standbyLabel, title, total }, {});

    expect(container).toMatchSnapshot();
  });

  it('should render points credit only in the standBy list footer component', () => {
    const {
      sameDayConfirmationPage: {
        fareSummary: {
          refundMessage,
          totalCredit,
          totalCredit: { item: title },
          taxesAndFeesWithLinks
        },
        _links: {
          viewBoardingPositions: { labelText: viewBoardingPositionsLabel }
        }
      }
    } = new sameDayPurchaseConfirmationBuilder().withOnlyPointsCredit().build();

    const { container } = createComponent(
      { isShowPoints: true, refundMessage, taxesAndFeesWithLinks, title, totalCredit, viewBoardingPositionsLabel },
      {}
    );

    expect(container).toMatchSnapshot();
  });

  it('should render a link with the standby search token when in web view', () => {
    const props = {
      isWebView: true,
      recordLocator: 'recordLocator',
      standbyLabel: 'standbyLabel',
      standbyToken: 'standbyToken'
    };
    const { queryByText } = createComponent(props);

    expect(queryByText('buttonText')).toMatchSnapshot();
  });

  it('should render a link with the boarding pass details when in web view', () => {
    const props = {
      buttonLabel: 'buttonLabel',
      firstName: 'First',
      lastName: 'Last',
      isWebView: true,
      recordLocator: 'recordLocator'
    };
    const { queryByText } = createComponent(props);

    expect(queryByText('buttonLabel')).toMatchSnapshot();
  });

  it('should render a link with the view boarding details with passenger search token when in web view', () => {
    const props = {
      firstName: 'John',
      isWebView: true,
      lastName: 'Appleseed',
      passengerSearchToken: 'abc.123.xyz',
      recordLocator: 'abc123',
      viewBoardingPositionsLabel: 'Boarding details'
    };
    const { getByText } = createComponent(props);

    expect(getByText('Boarding details')).toMatchSnapshot();
  });

  const createComponent = (props = {}, state = {}) => {
    const defaultProps = {
      location: {},
      match: { params: '' }
    };
    const defaultState = {
      app: {
        errorHeader: {
          errorMessage: null,
          hasError: false
        },
        sameDay: {}
      },
      router: {
        location: {
          search: 'search'
        }
      }
    };
    const mergedProps = { ...defaultProps, ...props };
    const store = configureMockStore()({ ...defaultState, ...state });

    return render(
      <div>
        <Provider store={store}>
          <StandByListFooter {...mergedProps} />
        </Provider>
      </div>
    );
  };
});
