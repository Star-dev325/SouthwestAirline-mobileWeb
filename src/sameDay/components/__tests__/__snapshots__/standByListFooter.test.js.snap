// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandByListFooter should  render standBy list footer component with all information 1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--price-total price-difference-footer--price-total bgblue"
      >
        <div
          class="price-line"
        >
          <span
            class="price-line--title"
          >
            Total Paid
          </span>
          <div
            class="price-amount"
          >
            <div
              class="price-line--currency_total-per-section"
            >
              <div
                class="currency"
              >
                <span
                  class="money-sign"
                >
                  $
                </span>
                <span
                  data-qa="total-amount"
                >
                  0.00
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total bgblue stand-by-list-footer--taxes-and-fees_reverse"
      >
        <div
          class="stand-by-list-footer stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            You will not be charged until you board new standby flight
          </p>
        </div>
      </div>
      <div
        class="early-bird-price-footer--nav stand-by-list-footer--nav bgblue"
      >
        <button
          class="button button--fluid larger button--blue"
          type="submit"
        >
          see standby list
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`StandByListFooter should not render explanatory note(taxesAndFeesWithLinks) in the standBy list footer component 1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--price-total price-difference-footer--price-total bgblue"
      >
        <div
          class="price-line"
        >
          <span
            class="price-line--title"
          >
            Total Paid
          </span>
          <div
            class="price-amount"
          >
            <div
              class="price-line--currency_total-per-section"
            >
              <div
                class="currency"
              >
                <span
                  class="money-sign"
                >
                  $
                </span>
                <span
                  data-qa="total-amount"
                >
                  0.00
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total bgblue stand-by-list-footer--taxes-and-fees_reverse"
      />
      <div
        class="early-bird-price-footer--nav stand-by-list-footer--nav bgblue"
      >
        <button
          class="button button--fluid larger button--blue"
          type="submit"
        >
          see standby list
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`StandByListFooter should not render standby list button in the standBy list footer component 1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--price-total price-difference-footer--price-total-with-rule bgblue"
      >
        <div
          class="price-line"
        >
          <span
            class="price-line--title"
          >
            Total Paid
          </span>
          <div
            class="price-amount"
          >
            <div
              class="price-line--currency_total-per-section"
            >
              <div
                class="currency"
              >
                <span
                  class="money-sign"
                >
                  $
                </span>
                <span
                  data-qa="total-amount"
                >
                  0.00
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total-with-rule bgblue"
      >
        <div
          class="stand-by-list-footer stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            You will not be charged until you board new standby flight
          </p>
        </div>
      </div>
      <div
        class="stand-by-list-footer--divider bgblue"
      >
        <hr
          class="stand-by-list-footer--divider-rule"
        />
      </div>
      <div
        class="stand-by-list-footer--btn price-difference-footer--price-total-with-rule bgblue"
      >
        <div>
          <button
            class="button button--fluid larger button--yellow"
            type="submit"
          >
            SHARED__BUTTON_TEXT__BOARDING_DETAILS
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`StandByListFooter should not render total price in the standBy list footer component 1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total bgblue stand-by-list-footer--taxes-and-fees_reverse"
      >
        <div
          class="stand-by-list-footer stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            You will not be charged until you board new standby flight
          </p>
        </div>
      </div>
      <div
        class="early-bird-price-footer--nav stand-by-list-footer--nav bgblue"
      >
        <button
          class="button button--fluid larger button--blue"
          type="submit"
        >
          see standby list
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`StandByListFooter should render Boarding details label in the standBy list footer component 1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--price-total price-difference-footer--price-total bgblue"
      >
        <div
          class="price-line"
        >
          <span
            class="price-line--title"
          >
            Total Credit
          </span>
          <div
            class="price-amount"
          >
            <div
              class="price-line--currency_total-per-section"
            >
              <div
                class="currency"
              >
                <span
                  class="money-sign"
                >
                  $
                </span>
                <span
                  data-qa="total-amount"
                >
                  20.00
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total bgblue bggreen stand-by-list-footer--taxes-and-fees_top stand-by-list-footer--taxes-and-fees_reverse"
      >
        <div
          class="stand-by-list-footer stand-by-list-footer--refund-taxes-note_right stand-by-list-footer--refund-taxes-note stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            Refunded to credit card
          </p>
        </div>
        <div
          class="stand-by-list-footer stand-by-list-footer--refund-taxes-note_right stand-by-list-footer--refund-taxes-note stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            Includes taxes and fees
          </p>
        </div>
      </div>
      <div
        class="stand-by-list-footer--btn_right price-difference-footer--price-total bgblue bggreen"
      >
        <div>
          <button
            class="button button--fluid larger button--yellow"
            type="submit"
          >
            Boarding details
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`StandByListFooter should render a link with the boarding pass details when in web view 1`] = `null`;

exports[`StandByListFooter should render a link with the standby search token when in web view 1`] = `null`;

exports[`StandByListFooter should render a link with the view boarding details with passenger search token when in web view 1`] = `
<a
  class="button button_link button--fluid larger button--yellow"
  href="app://boarding-details?firstName=John&lastName=Appleseed&passengerSearchToken=abc.123.xyz&recordLocator=abc123"
  target="_self"
  type="submit"
>
  Boarding details
</a>
`;

exports[`StandByListFooter should render points credit only in the standBy list footer component 1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total-with-rule bgblue bggreen stand-by-list-footer--taxes-and-fees_top stand-by-list-footer--taxes-and-fees_reverse"
      >
        <div
          class="stand-by-list-footer stand-by-list-footer--refund-taxes-note_right stand-by-list-footer--refund-taxes-note stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            Held for future use
          </p>
        </div>
        <div
          class="stand-by-list-footer stand-by-list-footer--refund-taxes-note_right stand-by-list-footer--refund-taxes-note stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            Included taxes and fees
          </p>
        </div>
      </div>
      <div
        class="stand-by-list-footer--divider bgblue bggreen"
      >
        <hr
          class="stand-by-list-footer--divider-rule"
        />
      </div>
      <div
        class="stand-by-list-footer--btn price-difference-footer--price-total-with-rule bgblue bggreen"
      >
        <div>
          <button
            class="button button--fluid larger button--yellow"
            type="submit"
          >
            Boarding details
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`StandByListFooter should render standby list button with extra bottom padding in the standBy list footer component when isWebView 1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total bgblue bggreen stand-by-list-footer--taxes-and-fees_top stand-by-list-footer--taxes-and-fees_reverse"
      >
        <div
          class="stand-by-list-footer stand-by-list-footer--refund-taxes-note_right stand-by-list-footer--refund-taxes-note stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            Refunded to credit card
          </p>
        </div>
        <div
          class="stand-by-list-footer stand-by-list-footer--refund-taxes-note_right stand-by-list-footer--refund-taxes-note stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            You will not be refunded until you board, and new flight is closed.
          </p>
        </div>
      </div>
      <div
        class="early-bird-price-footer--nav stand-by-list-footer--nav bgblue bggreen stand-by-list-footer--nav_webview"
      >
        <a
          class="button button_link button--fluid larger button--blue"
          href="app://standby?recordLocator=undefined&standbyToken=undefined"
          target="_self"
          type="submit"
        >
          see standby list
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`StandByListFooter should render taxCreditRefund with points credit and tax credit in the standBy list footer component 1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--price-total price-difference-footer--price-total bggreen"
      >
        <div
          class="price-line"
        >
          <span
            class="price-line--title"
          >
            Total Credit
          </span>
          <div
            class="price-amount"
          >
            <div
              class="price-line--currency_total-per-section"
            >
              <div
                class="currency"
              >
                <span
                  class="money-sign"
                >
                  $
                </span>
                <span
                  data-qa="total-amount"
                >
                  5.60
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="stand-by-list-footer--taxes-and-fees_top stand-by-list-footer--taxes-and-fees_reverse"
        >
          <div>
            <p
              class="stand-by-list-footer--price-refund-message"
            >
              Refunded to credit card
            </p>
          </div>
          <div>
            <p
              class="stand-by-list-footer--price-refund-message stand-by-list-footer--price-refund-message_top-margin"
            >
              Included taxes and fees
            </p>
          </div>
        </div>
      </div>
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total bggreen stand-by-list-footer--taxes-and-fees_top stand-by-list-footer--taxes-and-fees_reverse"
      />
      <div
        class="early-bird-price-footer--nav stand-by-list-footer--nav bgblue bggreen"
      >
        <button
          class="button button--fluid larger button--blue"
          type="submit"
        >
          Boarding details
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`StandByListFooter should render total credit and standby list button in the standBy list footer component  1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total bgblue bggreen stand-by-list-footer--taxes-and-fees_top stand-by-list-footer--taxes-and-fees_reverse"
      >
        <div
          class="stand-by-list-footer stand-by-list-footer--refund-taxes-note_right stand-by-list-footer--refund-taxes-note stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            Refunded to credit card
          </p>
        </div>
        <div
          class="stand-by-list-footer stand-by-list-footer--refund-taxes-note_right stand-by-list-footer--refund-taxes-note stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            You will not be refunded until you board, and new flight is closed.
          </p>
        </div>
      </div>
      <div
        class="early-bird-price-footer--nav stand-by-list-footer--nav bgblue bggreen"
      >
        <button
          class="button button--fluid larger button--blue"
          type="submit"
        >
          see standby list
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`StandByListFooter should render total with points and no tax in the standBy list footer component 1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--price-total price-difference-footer--price-total bgblue"
      >
        <div
          class="price-line"
        >
          <span
            class="price-line--title"
          >
            Total Paid
          </span>
          <div
            class="price-amount"
          >
            <div
              class="price-line--currency_total-per-section"
            >
              <div
                class="currency"
              >
                <span
                  data-qa="total-amount"
                >
                  3,400
                </span>
                <span
                  class="points-sign"
                >
                  SHARED__COMMON__POINTS_TEXT
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total bgblue stand-by-list-footer--taxes-and-fees_reverse"
      >
        <div
          class="stand-by-list-footer stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            You will not be charged until you board new standby flight
          </p>
        </div>
      </div>
      <div
        class="early-bird-price-footer--nav stand-by-list-footer--nav bgblue"
      >
        <button
          class="button button--fluid larger button--blue"
          type="submit"
        >
          see standby list
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`StandByListFooter should render total with points and tax in the standBy list footer component 1`] = `
<div>
  <div>
    <div
      class="stand-by-list-footer"
    >
      <div
        class="stand-by-list-footer--price-total price-difference-footer--price-total bgblue"
      >
        <div
          class="price-line"
        >
          <span
            class="price-line--title"
          >
            Total Paid
          </span>
          <div
            class="price-amount"
          >
            <div
              class="price-line--currency_total-per-section"
            >
              <div
                class="currency"
              >
                <span
                  data-qa="total-amount"
                >
                  3,400
                </span>
                <span
                  class="points-sign"
                >
                  SHARED__COMMON__POINTS_TEXT
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="stand-by-list-footer--price-total price-difference-footer--price-total bgblue"
      >
        <div
          class="price-line"
        >
          <span
            class="price-line--title"
          >
            Total Paid
          </span>
          <div
            class="price-amount"
          >
            <div
              class="price-line--currency_total-per-section"
            >
              <div
                class="currency"
              >
                <span
                  class="money-sign"
                >
                  $
                </span>
                <span
                  data-qa="total-amount"
                >
                  5.60
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="stand-by-list-footer--taxes-and-fees price-difference-footer--price-total bgblue stand-by-list-footer--taxes-and-fees_reverse"
      >
        <div
          class="stand-by-list-footer stand-by-list-footer--sub-note"
        >
          <p
            class="white"
          >
            You will not be charged until you board new standby flight
          </p>
        </div>
      </div>
      <div
        class="early-bird-price-footer--nav stand-by-list-footer--nav bgblue"
      >
        <button
          class="button button--fluid larger button--blue"
          type="submit"
        >
          see standby list
        </button>
      </div>
    </div>
  </div>
</div>
`;
